imblearn/utils/_validation.py

```diff
@@ -27,6 +27,42 @@
 TARGET_KIND = ("binary", "multiclass", "multilabel-indicator")
 
 
+class OutputFormater:
+    """A class for converting input types to numpy and back."""
+
+    def __init__(self, X, y):
+        self.x_props = self._gets_props(X)
+        self.y_props = self._gets_props(y)
+
+    def format(self, X, y):
```

Comment:
Maybe transform instead of format

Reply:
Yes. I changed this to be inline with the class name. I will change it accordingly.